### PR TITLE
feat(healthcheck): Add support to add healthcheck per proctype

### PR DESCRIPTION
**Pull request by kmala**

### CODE REVIEW

File: api/config.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type Config struct {
 	// (ex 1) or in milli units to reflect the number of CPU shares (ex 500m).
 	CPU map[string]interface{} `json:"cpu,omitempty"`
 	// Healthchecks are the healthchecks that the application uses.
-	Healthcheck map[string]*Healthcheck `json:"healthcheck,omitempty"`
+	Healthcheck map[string]*Healthchecks `json:"healthcheck,omitempty"`
```
Comment (Joshua-Anderson) [inline api/config.go]: Is the map formatted like `map[<process>][<healthcheck type>]`?

Is so, could it possibly be documented in the comment above for people using the API?

Comment (kmala) [inline api/config.go]: i had written below for Healthchecks and assumed it would convey...i can mention it here also

**Comment by Joshua-Anderson** [comment]
This needs to be rebased for CI to pass.

File: api/config.go (hunk 1 above)
Comment (Joshua-Anderson) [inline api/config.go]: You documented the second map field, but it is still ambiguous what the field here represents.

Comment (kmala) [inline api/config.go]: Updated.

File: api/config.go [hunk 2]
```diff
@@ -52,6 +52,10 @@ type ConfigHookRequest struct {
 	App  string `json:"receive_repo"`
 }
 
+// Healthchecks is a map of different Healthcheck's that a process of application uses.
```
Comment (mboersma) [inline api/config.go]: Typo: should be "Healthchecks" without an apostrophe `'`. Also I think this should be reworded a bit:
> // Healthchecks maps an app process name to a healthcheck type.
> // The valid values are "livenessProbe" and "readinessProbe".

Comment (kmala) [inline api/config.go]: Updated.

**Pull request merged**